internal/transformers/declarations/transform.go

```diff
@@ -1767,3 +1772,12 @@ func (tx *DeclarationTransformer) transformJSDocOptionalType(input *ast.JSDocOpt
 	tx.EmitContext().SetOriginal(replacement, input.AsNode())
 	return replacement
 }
+
+func (tx *DeclarationTransformer) getGeneratedDefaultExportIdentifier() *ast.Node {
```

Comment:
This function feels wrong to me; this means we're putting the same node into the tree multiple times, which is a no-no.

Reply:
(I'm not 100% sure this fix is required for the rest of the PR, so would probably be worth a follow-up as a d.ts improvement?)

Reply:
Ah, ye - I meant to ask if I should clone this before returning and forgot. 

> (I'm not 100% sure this fix is required for the rest of the PR, so would probably be worth a follow-up as a d.ts improvement?)

yes, it's a little bit out of scope of this PR. This has just popped up in the diffs after EOF got included. I'll remove it from this PR